packages/import/src/CreateCsv.ts

```diff
@@ -135,21 +151,67 @@ export class CreateCsv {
     }
 
     async run(params: CsvGenerationParams) {
-        const { numOfRows } = params;
         const csvStream = csv.format({ headers: true });
         const fileStream = createWriteStream("items.csv");
         csvStream.pipe(fileStream);
+        await this.generateCsv(params, (row) => csvStream.write(row))
+        csvStream.end();
+    }
+
+    async generateCsv(params: CsvGenerationParams, writer: (row: Row) => void) {
+        const { withFile } = params;
+        let numOfRows:number = params.numOfRows ? params.numOfRows : 1;
         let numOfColumns = 0;
+        let filesNames: string[] | undefined = undefined;
+        if (withFile && withFile.dir) {
+            filesNames = this.readFileNames(withFile.dir, withFile.contentType);
+            numOfRows = filesNames.length;
+        }
         for (let i = 0; i < numOfRows; ++i) {
-            const row = this.createRow(params, i);
+            let file: File | undefined = undefined;
+            if (withFile) {
+                file = filesNames ?
+                    await this.readFile(withFile, filesNames[i]) :
+                    this.scaffold(withFile.contentType, i)
+            }
+            const row = await this.createRow(params, file, i);
+            writer(row);
             numOfColumns = Object.keys(row).length;
-            csvStream.write(row);
         }
-        csvStream.end();
         console.log(`items.csv generated with ${ numOfRows } row(s) and ${ numOfColumns } columns`)
     }
 
-    createRow(params: RowGenerationParams, i = 0): Row {
+    readFileNames(dir: string, mimeType: MimeType): string[] {
+        return fs.readdirSync(dir).filter(file => {
```

Comment:
There is a promise-based API for reading the content of a directory: https://nodejs.org/api/fs.html#fspromisesreaddirpath-options

Reply:
In this particular context (a CLI), it is indeed OK. However, by using the sync API, you block the event loop while performing I/O, which makes it unavailable for other background tasks. But again, you are right, it will not change anything here.